homeassistant/components/azure_event_hub/config_flow.py

```diff
@@ -0,0 +1,208 @@
+"""Config flow for azure_event_hub integration."""
+from __future__ import annotations
+
+from copy import deepcopy
+import logging
+from typing import Any
+
+from azure.eventhub.exceptions import EventHubError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .client import AzureEventHubClient
+from .const import (
+    CONF_EVENT_HUB_CON_STRING,
+    CONF_EVENT_HUB_INSTANCE_NAME,
+    CONF_EVENT_HUB_NAMESPACE,
+    CONF_EVENT_HUB_SAS_KEY,
+    CONF_EVENT_HUB_SAS_POLICY,
+    CONF_MAX_DELAY,
+    CONF_SEND_INTERVAL,
+    CONF_USE_CONN_STRING,
+    DEFAULT_OPTIONS,
+    DOMAIN,
+    STEP_CONN_STRING,
+    STEP_SAS,
+    STEP_USER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EVENT_HUB_INSTANCE_NAME): str,
+        vol.Optional(CONF_USE_CONN_STRING, default=False): bool,
+    }
+)
+
+CONN_STRING_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EVENT_HUB_CON_STRING): str,
+    }
+)
+
+SAS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EVENT_HUB_NAMESPACE): str,
+        vol.Required(CONF_EVENT_HUB_SAS_POLICY): str,
+        vol.Required(CONF_EVENT_HUB_SAS_KEY): str,
+    }
+)
+
+
+async def validate_data(data: dict[str, Any]) -> dict[str, str] | None:
+    """Validate the input."""
+    client = AzureEventHubClient.from_input(**data)
+    try:
+        await client.test_connection()
+    except EventHubError:
+        return {"base": "cannot_connect"}
+    except Exception as exc:  # pylint: disable=broad-except
+        _LOGGER.warning("Unknown error: %s", exc)
```

Comment:
We normally log with stack trace for unexpected exceptions in the flow.
```suggestion
        _LOGGER.exception("Unknown error")
```